backend/instancemgmt/instance_manager.go

```diff
@@ -156,3 +156,19 @@ func callInstanceHandlerFunc(fn InstanceCallbackFunc, instance interface{}) {
 	params = append(params, reflect.ValueOf(instance))
 	reflect.ValueOf(fn).Call(params)
 }
+
+func (c CachedInstance) IsStale(pluginContext backend.PluginContext) bool {
```

Comment:
Not sure about this refactoring since it combines separate logic (ds vs app) in the same place. Is this kind of what you pictured @andresmgot?

Reply:
Yeah while there's some duplicate logic, I would rather keep them separate I think. Thanks for chiming in @xnyo

Reply:
This method is not in use currently right?

Reply:
Good spot! Removed ❌